system/lib/wasmfs/file.h

```diff
@@ -30,7 +30,9 @@ const backend_t NullBackend = nullptr;
 
 class File : public std::enable_shared_from_this<File> {
 public:
-  enum FileKind { DataFileKind = 0, DirectoryKind, SymlinkKind };
+  enum FileKind { UnknownKind, DataFileKind, DirectoryKind, SymlinkKind };
```

Comment:
Why not reuse DT_CHR and friends here?

Reply:
Mostly for readability, but also to make it more explicit that we only support a subset of the available file types.